src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/conditions/ElectronicMonitoringProgrammeService.kt

```diff
@@ -0,0 +1,119 @@
+  private val electronicMonitoringProviderRepository: ElectronicMonitoringProviderRepository,
+) {
+
+  @Transactional
+  fun updateElectronicMonitoringProgramme(licenceId: Long, request: UpdateElectronicMonitoringProgrammeRequest) {
+    val licenceEntity = licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val username = SecurityContextHolder.getContext().authentication.name
+    val staffMember = staffRepository.findByUsernameIgnoreCase(username)
+
+    val electronicMonitoringProvider = when (licenceEntity) {
+      is CrdLicence -> requireNotNull(licenceEntity.electronicMonitoringProvider) {
+        "ElectronicMonitoringProvider is null for CrdLicence: $licenceId"
+      }
+
+      is HdcLicence -> requireNotNull(licenceEntity.electronicMonitoringProvider) {
+        "ElectronicMonitoringProvider is null for HdcLicence: $licenceId"
+      }
+
+      else -> error("Trying to update electronic monitoring provider details for non-crd or non-hdc: $licenceId")
+    }
+    electronicMonitoringProvider.isToBeTaggedForProgramme = request.isToBeTaggedForProgramme
+    electronicMonitoringProvider.programmeName = request.programmeName
+    licenceRepository.saveAndFlush(licenceEntity)
+
+    auditService.recordAuditEventUpdateElectronicMonitoringProgramme(licenceEntity, request, staffMember)
+  }
+
+  @Transactional
+  fun handleElectronicMonitoringResponseRecords(licenceEntity: Licence) {
+    val additionalApConditions = licenceEntity.additionalConditions.filter { it.conditionType == "AP" }
+    val conditionsRequiringElectronicMonitoringResponse = licencePolicyService.getConditionsRequiringElectronicMonitoringResponse(licenceEntity.version!!, additionalApConditions.map { it.conditionCode })
+    if (conditionsRequiringElectronicMonitoringResponse.isNotEmpty()) {
+      LicenceConditionService.log.info("Handling Electronic Monitoring response record for conditions: ${conditionsRequiringElectronicMonitoringResponse.joinToString(",") { it.code }}")
+      initialiseElectronicMonitoringProviderIfNotExists(licenceEntity)
+    } else {
+      clearElectronicMonitoringProvider(licenceEntity)
+    }
+  }
+
+  @Transactional
+  fun clearElectronicMonitoringProvider(licenceEntity: Licence) {
+    LicenceConditionService.log.info("Clearing Electronic Monitoring response records for licence: ${licenceEntity.id}")
+    when (licenceEntity) {
+      is CrdLicence -> {
+        licenceEntity.electronicMonitoringProvider?.let {
+          electronicMonitoringProviderRepository.delete(it)
+        }
+        licenceEntity.electronicMonitoringProvider = null
+      }
+      is HdcLicence -> {
+        licenceEntity.electronicMonitoringProvider?.let {
+          electronicMonitoringProviderRepository.delete(it)
+        }
+        licenceEntity.electronicMonitoringProvider = null
+      }
+    }
+  }
+
+  @Transactional
+  fun initialiseElectronicMonitoringProviderIfNotExists(licenceEntity: Licence) {
```

Comment:
sorry to be a pain but create is more appropriate, you would no expect a dyslexic to be fussy about words :-), it's just I strongly believe in self describing code